src/snowflake/connector/nanoarrow_cpp/ArrowIterator/FixedSizeListConverter.cpp

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+//
+
+#include "FixedSizeListConverter.hpp"
+
+namespace sf
+{
+Logger* FixedSizeListConverter::logger =
+    new Logger("snowflake.connector.FixedSizeListConverter");
+
+FixedSizeListConverter::FixedSizeListConverter(ArrowArrayView* array)
+: m_array(array)
+{
+}
+
+void FixedSizeListConverter::generateError(const std::string& msg) const
+{
+  logger->error(__FILE__, __func__, __LINE__, msg.c_str());
+  PyErr_SetString(PyExc_Exception, msg.c_str());
+}
+
+PyObject* FixedSizeListConverter::toPyObject(int64_t rowIndex) const
+{
+  if (ArrowArrayViewIsNull(m_array, rowIndex))
+  {
+    Py_RETURN_NONE;
+  }
+
+  if (m_array->n_children != 1)
+  {
+    std::string errorInfo = Logger::formatString(
+        "[Snowflake Exception] invalid arrow element schema for fixed size "
+        "list: got (%d) "
+        "children",
+        m_array->n_children);
+    this->generateError(errorInfo);
+    return nullptr;
+  }
+
+  // m_array->length represents the number of fixed size lists in the array
+  // m_array->children[0] has a buffer view that contains the actual data of
+  // each list, back-to-back m_array->children[0]->length represents the sum of
+  // the lengths of the fixed size lists in the array.
+
+  ArrowArrayView* elements = m_array->children[0];
+  const auto length = elements->length / m_array->length;
```

Comment:
nit: can we rename this to `fixedSizeArrayLength`? There are a lot of lengths already so let's not introduce yet another `length` :p